docs: Improve BGP Control Plane page (#23939)
cilium-team/cilium: Documentation/network/bgp-control-plane.rst

   --enable-bgp-control-plane=true

> Following values have to be set for helm installation

Reviewer: OK, I'd maybe rephrase like this:

```suggestion
If using Helm charts instead, the relevant values are the following:
```

Nit: You also have a typo in the commit description for this change (`arument` -> `argument`).

.. code-block:: yaml

   bgpControlPlane:
     enabled: true

.. note::

>    The BGP Control Plane feature is mutually exclusive with the :ref:`bgp`
>    feature and therefore it has to be disabled.
>    If both are enabled, the BGP will be prioritized and will probably fail.
>    In other words, this feature does not switch the BGP implementation
>    from MetalLB to GoBGP.

Reviewer: This is a useful note, but is a bit confusing at times (it's hard to understand which BGP implementation we're talking about, and GoBGP/MetalLB aren't otherwise introduced in the page). Also if one has to be disabled anyway, I wouldn't try guessing what happens when both are enabled, let's just stick to “one has to be off”. Some suggestions below:

```suggestion
   The BGP Control Plane feature is mutually exclusive with the MetalLB-based :ref:`bgp`
   feature. To use the Control Plane, the older BGP feature has to be disabled.
   In other words, this feature does _not_ switch the BGP implementation
   from MetalLB to GoBGP.
```
